network/test/middleware_test.go

```diff
@@ -805,6 +811,80 @@ func (m *MiddlewareTestSuite) TestUnsubscribe() {
 	unittest.RequireNeverReturnBefore(m.T(), msgRcvdFun, 2*time.Second, "message received unexpectedly")
 }
 
+// TestUnicastStreamLimit tests that the middleware does not accept more than the maximum number of
+// unicast streams and streams created beyond the limit are reset
+func (m *MiddlewareTestSuite) TestUnicastStreamLimit() {
+	sender := 0
+	receiver := m.size - 1
+
+	senderNodeID := m.ids[sender].NodeID
+	receiverNodeID := m.ids[receiver].NodeID
+
+	senderPeerID := m.nodes[sender].Host().ID()
+	receiverPeerID := m.nodes[receiver].Host().ID()
+
+	// configured mw.unicastMaxStreamsPerPeer (using default)
+	maxStreams := middleware.DefaultMaxUnicastStreamsPerPeer
+
+	receiveCount := 0
+	m.ov[receiver].On("Receive", senderNodeID, mockery.Anything, mockery.Anything).Return(nil).Run(func(_ mockery.Arguments) {
+		receiveCount++
+	})
+
+	var protocol protocol.ID
```

Comment:
```suggestion
	var streamProtocol protocol.ID
```
`protocol` collides with a package name.